# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentChild.h
# Commit: 1bcf9f586c55
# Full Hash: 1bcf9f586c5582ddeb822f4c2be3ca314694337e
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 04:41:59
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 11 - Use RemoteBrowser interface instead of IPDL actors in nsFrameLoader. r=nika
#   
#   This commit replaces the direct use of the IPDL actors in nsFrameLoader with
#   the RemoteBrowser interface. Some special use cases are adapted to still use
#   the IPDL actors. In the future, we should burn these use cases down.
# ==============================================================================

diff -r d3b2ac8d5ca4 -r 1bcf9f586c55 dom/ipc/ContentChild.h
--- a/dom/ipc/ContentChild.h	Wed May 15 10:34:03 2019 -0500
+++ b/dom/ipc/ContentChild.h	Mon May 06 21:29:48 2019 -0500
@@ -13,6 +13,7 @@
 #include "mozilla/dom/BrowserBridgeChild.h"
 #include "mozilla/dom/PBrowserOrId.h"
 #include "mozilla/dom/PContentChild.h"
+#include "mozilla/dom/RemoteBrowser.h"
 #include "mozilla/dom/CPOWManagerGetter.h"
 #include "mozilla/StaticPtr.h"
 #include "mozilla/ipc/Shmem.h"
@@ -162,7 +163,7 @@
 
   static void UpdateCookieStatus(nsIChannel* aChannel);
 
-  static already_AddRefed<BrowserBridgeChild> CreateBrowser(
+  static already_AddRefed<RemoteBrowser> CreateBrowser(
       nsFrameLoader* aFrameLoader, const TabContext& aContext,
       const nsString& aRemoteType, BrowsingContext* aBrowsingContext);
 